src/model/line/mod.rs

```diff
@@ -26,7 +28,11 @@ impl Line {
             "C4 F3 G3 Bb3 C4 Db4 Bb3 Db4 C4 . .",
         ]
         .iter()
-        .map(|line_str| parse_line(line_str).unwrap())
+        .map(|line_str| Self::parse(line_str).unwrap())
         .collect()
     }
+
+    pub fn parse(string: &str) -> Result<Self> {
```

Comment:
Looking at the `parser()` method pattern you introduced, it felt to me like (a) there should probably be another level of convenience to that via a corresponding `parse()` method and (b) it seemed like the existing `line_parser` module was basically just a separate module implementing that functionality for `Line`, so moved it to `model/line/parser.rs` and exposed it as `Line::parse()`